dynamic_programming/house_robber.cpp

```diff
@@ -0,0 +1,108 @@
+/**
+ * @file
+ * @brief Implementation of [House Robber Problem] algorithm
+ * @details
+ * Solution of House robber problem uses a dynamic programming concept that
+ * works in \f$O(n)\f$ time and works in \f$O(1)\f$ space.
+ * @author [Swastika Gupta](https://github.com/swastyy)
+ */
+
+#include <cassert>   /// for assert
+#include <climits>   /// for std::max
+#include <iostream>  /// for io operations
+#include <vector>    /// for std::vector
+/**
+ * @namespace dynamic_programming
+ * @brief House robber algorithm
+ */
+namespace dynamic_programming {
+/**
+ * @namespace house_robber
+ * @brief Functions for House Robber algorithm
+ */
+namespace house_robber {
+/**
+ * @brief The main function implements house robber algorithm using dynamic
+ * programming
+ * @param money array containing money in the ith house
+ * @param n size of array
+ * @returns maximum amount of money that can be robbed
+ */
+std::uint64_t houseRobber(const std::vector<int> &money, int n) {
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). 🙂

Reply:
Since the input array denoting money in the ith house is non-negative, the least amount of money that can be robbed is zero.

Reply:
You could use `uint32_t` for that (as `uint`-based types are used for non-negative values, while `int`-based types (for example `int64_t`) are used for negative values).

Reply:
Ohh Ohkk 👍